cmd/hook.go

```diff
@@ -376,7 +376,12 @@ Gitea or set your environment appropriately.`, "")
 		oldCommitIDs[count] = string(fields[0])
 		newCommitIDs[count] = string(fields[1])
 		refFullNames[count] = git.RefName(fields[2])
-		if refFullNames[count] == git.BranchPrefix+"master" && newCommitIDs[count] != git.EmptySHA && count == total {
+
+		hash, err := git.HashFromString(newCommitIDs[count])
+		if err != nil {
+			return err
+		}
+		if refFullNames[count] == git.BranchPrefix+"master" && newCommitIDs[count] != hash.Type().Empty().String() && count == total {
```

Comment:
Here and below, it makes the code more complex.

Could it be simplified to something like: 

```
if git.IsEmptyXxxx()
```

?

Reply:
I guess a helper function? Would it have error or just return false on error? For example, isEmptyHash("000") == false? since it's not correct length?

I now also tried a different approach and try to determine the hash type from repository path via passing EmptyHash to some git command (it errors when repo is of different type). But that's a little heavy.